### PR TITLE
[ONNX] Export aten::native_dropout

### PR DESCRIPTION
**Pull request by wschin**
This PR adds exporter for `aten::native_dropout` with a test.

### CODE REVIEW

File: torch/onnx/symbolic_opset12.py [hunk 1]
```diff
@@ -48,15 +48,21 @@ def outer(g, input, other):
...
     symbolic_helper.check_training_mode(train, "dropout")
-    # if train is False, dropout is no-op
+    # in eval mode, dropout is non-op - if the node's train param is set to False, dropout is non-op
     if not train:
         return input
     p = g.op("Constant", value_t=torch.tensor(p))
-    t = g.op("Constant", value_t=torch.tensor(train, dtype=torch.bool))
-    r, _ = g.op("Dropout", input, p, t, outputs=2)
-    return r
+    t = g.op("Constant", value_t=torch.tensor(True))
```
Comment (wschin) [inline torch/onnx/symbolic_opset12.py:57]:
```suggestion
    t = g.op("Constant", value_t=torch.tensor(train, dtype=torch.bool))
```

**Review by BowenBao — APPROVED** [review]

File: torch/onnx/symbolic_opset12.py [hunk 2]
```diff
@@ -47,16 +47,26 @@ def outer(g, input, other):
     return einsum_helper(g, "i,j->ij", [input, other])
 
 
-@symbolic_helper.parse_args("v", "f", "i")
-def dropout(g, input, p, train):
+def _dropout_shared(g, input, p, train):
     symbolic_helper.check_training_mode(train, "dropout")
-    # if train is False, dropout is no-op
+    # in eval mode, dropout is non-op - if the node's train param is set to False, dropout is non-op
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset12.py:52]: no-op: https://en.wikipedia.org/wiki/NOP_(code)

File: torch/onnx/symbolic_opset12.py [hunk 3]
```diff
@@ -47,16 +47,26 @@ def outer(g, input, other):
     return einsum_helper(g, "i,j->ij", [input, other])
 
 
-@symbolic_helper.parse_args("v", "f", "i")
-def dropout(g, input, p, train):
+def _dropout_shared(g, input, p, train):
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset12.py:50]: I would annotate the type

Comment (justinchuby) [inline torch/onnx/symbolic_opset12.py:50]: Naming: Consider a more descriptive name for the function?

Comment (justinchuby) [inline torch/onnx/symbolic_opset12.py:50]:
```suggestion
def _dropout_shared(g: _C.Graph, input: _C.Value, p: float, train: bool) -> Tuple[_C.Value, Optional[_C.Value]]:
```

assuming mask is a Value?

**Review by justinchuby — CHANGES_REQUESTED** [review]
nit, thanks!

File: torch/onnx/symbolic_opset12.py (hunk 3 above)
Comment (wschin) [reply torch/onnx/symbolic_opset12.py:50]: It will throw, so no exporter function does that.

Comment (wschin) [reply torch/onnx/symbolic_opset12.py:50]: Not sure what will be more descriptive. `_dropout_core`? `_shared_dropout_exporter`? `_dropout_common_part`? They all look similar to `_dropout_shared`...

Comment (wschin) [reply torch/onnx/symbolic_opset12.py:50]: As I mentioned in another comment, no exporter does type annotation because it will trigger an exception in CIs. Someone needs to fix the root cause. You can reproduce it by running
```
lintrunner -a master
```
and it will error out with
```
  Error (MYPY) [attr-defined]
    "Graph" has no attribute "op"
```

File: torch/onnx/symbolic_opset12.py (hunk 2 above)
Comment (wschin) [reply torch/onnx/symbolic_opset12.py]: Sure.

File: torch/onnx/symbolic_opset12.py (hunk 3 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset12.py]: What is the error you see?

Comment (justinchuby) [reply torch/onnx/symbolic_opset12.py]: We are working on type annotating more of our codebase. Let’s work together to see if we can do anything in this particular case

Comment (justinchuby) [reply torch/onnx/symbolic_opset12.py]: Let’s omit graph and annotate the rest for now?

Comment (wschin) [reply torch/onnx/symbolic_opset12.py]: Ok. This is the new signature:
```python
def _dropout_returns_masked_input_and_mask(
    g, input: torch._C.Value, p: float, train: bool
):
```

Comment (justinchuby) [reply torch/onnx/symbolic_opset12.py]: _C.Value is good, for consistency

File: torch/onnx/symbolic_opset12.py [hunk 4]
```diff
@@ -47,16 +47,29 @@ def outer(g, input, other):
     return einsum_helper(g, "i,j->ij", [input, other])
 
 
-@symbolic_helper.parse_args("v", "f", "i")
-def dropout(g, input, p, train):
+def _dropout_returns_masked_input_and_mask(
+    g, input: torch._C.Value, p: float, train: bool
+):
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset12.py:52]:
```suggestion
) -> Tuple[_C.Value, Optional[_C.Value]]:
```

File: torch/onnx/symbolic_opset12.py (hunk 3 above)
Comment (wschin) [reply torch/onnx/symbolic_opset12.py]: Hopefully you can also enforce/improve documentation on new/existing components. Debugging Pytorch is not easy. I remember calling `torch.add` results 40+ functions in the call stack, and I just don't know which function went wrong...

**Review by justinchuby — APPROVED** [review]

File: torch/onnx/symbolic_opset12.py [hunk 5]
```diff
@@ -47,16 +48,29 @@ def outer(g, input, other):
...
+
+
+@symbolic_helper.parse_args("v", "f", "i")
+def dropout(g, input, p, train):
+    masked, _ = _dropout_returns_masked_input_and_mask(g, input, p, train)
+    return masked
+
+
+@symbolic_helper.parse_args("v", "f", "i")
+def native_dropout(g, input, p, train):
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset12.py:72]: While you are at it, please help us type annotate these two symbolic functions as well. thanks!
